### ISSUE TITLE
TSC not watching for modification of tsconfig.json

### ISSUE THREAD

**Issue by adrianvoica** [Microsoft/TypeScript#11442]
**TypeScript Version:**  2.0.3

When in watch mode (`"watch": true` in **tsconfig.json**), tsc does not watch for tsconfig.json changes. 
Is this intended behavior? 
If so, what would be the reasoning behind this?

Thanks.

**Expected behavior:**
Compiler should reload the config file after changes (and restart compilation).

**Actual behavior:** 
Compiler ignores **tsconfig.json** modifications while in watch mode.

**Comment by basarat** [comment]
> Is this intended behavior

Yes. Its only watching the ts files.

**Comment by jmlopez-rod** [comment]
@adrianvoica so say I have my `tsconfig.json` file with `"watch": true` and then I start `tsc`. `tsc` will be watching files for file changes. Now say I modify the `tsconfig.json` file and I set watch to false. Do you want it to restart compilation but this time to not watch anymore?  I just don't think that's the scope of `tsc` or any other tool that has a configuration, say like `gulp` or `webpack`. If you modify its configuration then you have to manually restart it.

**Comment by adrianvoica** [comment]
@jmlopez-rod I think removing the watch mode (while not requiring to stop compilation) from  tsconfig.json is more of a corner case than say, adding new files in the "files" array in tsconfig (that's when I would need / want tsc to restart compilation). And regarding e.g.: gulp, not watching over itself (gulpfile) is a side effect, it's a limitation of its architecture, you would need to do some spawning workaround to achieve this.

Anyway, let's close this.

**Comment by MicahZoltu** [comment]
Chiming in here as this just bit me.  I have a watch running in VSCode and expect that as I change things about my project errors list will update.  After making a few changes to code/tsconfig.json, I noticed I had some outstanding errors around `console` being not found.  I messed with my `tsconfig.json` (changed `lib` compiler option) and when I didn't see my errors go away I went and spent another 10 minutes googling for answers (all of which told me to fix various compiler options).  Finally after a bunch of floundering I gave up and went to delete the offending lines and as soon as I touched the file all my errors in that file went away.  Luckily, I happened to leave my `tsconfig.json` in a state that fixed the errors when I touched the file, but it easily could have gone the other way.

As a user, I was surprised when changing my compiler options didn't result in my project being recompiled.  Of all the things to watch, this seems like one of the most relevant ones.  I agree with @adrianvoica that changing the `watch` compiler option is an edge case and as a user I would not be particularly surprised with any behavior around that.  Personally, I would go with whatever results in the least complexity of the engine, which I would *guess* means doing one last recompile with watch off.
